src/System.Private.CoreLib/src/System/Runtime/InteropServices/PInvokeMarshal.cs

```diff
@@ -17,7 +17,11 @@ public sealed class PInvokeMarshal
 
         public static void SaveLastWin32Error()
         {
+#if !PLATFORM_UNIX
             s_lastWin32Error = Interop.mincore.GetLastError();
```

Comment:
The preferred convention is to use partial classes + separate files for OS specific stuff instead of ifdefs.

From https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/cross-platform-guidelines.md

Partial classes is the approach we're currently taking when the managed code needs to diverge based on underlying platform